gdal/src/main/scala/geotrellis/gdal/GDALWarpOptions.scala

```diff
@@ -71,9 +71,9 @@ case class GDALWarpOptions(
   /** -te, set georeferenced extents of output file to be created (with a CRS specified) */
   te: Option[(Extent, CRS)] = None,
   /** -srcnodata, set nodata masking values for input bands (different values can be supplied for each band) */
-  srcNoData: List[Double] = Nil,
+  srcNoData: List[String] = Nil,
   /** -srcnodata, set nodata masking values for output bands (different values can be supplied for each band) */
-  dstNoData: List[Double] = Nil,
```

Comment:
This makes sense because at least GeoTiff stores NODATA as a `String`, but makes it a lot less specific for our needs. What was the value that caused the failure and what kind of failure was it? Potentially there are at least two other solutions: using `Double.NaN` or `List.empty` for cases when NODATA is not present on underlying `Dataset`.

Reply:
The errors were always caused by differing `NoData` values. From what I've seen, it looks like `NaN` is needed for `Float` and `Double`, and `List.empty` can be used for all other data types. Though, I'm not really sure if this is right. I'll need to look into it more to see what actually needs to be passed in.